tracker/dmlc_tracker/local.py

```diff
@@ -8,15 +8,43 @@
 import logging
 from threading import Thread
 from . import tracker
+from .opts import parse_env_pairs
 
-def exec_cmd(cmd, role, taskid, pass_env):
+def prepare_envs(args):
+    """
+    Load environment variables from arguments
+    """
+    envs = {}
+    # envs['default'] refers to the default env variables passed for both worker and server
+
+    # if --env option was used to pass a variable it would automatically be
+    # included because of the next line
+    envs['default'] = os.environ.copy()
```

Comment:
If `--env` is specified in the local tracker, we may want to add new environment variables. So
```
VAR1=VAL1 VAR2=VAL2 ./tracker/dmlc_tracker/dmlc-submit --cluster=local --env VAR3:VAL3
```
should assign `{'VAR1': 'VAL1', 'VAR2': 'VAL2', 'VAR3': 'VAL3'}` to `envs['default']`.

Reply:
Right now, it appears that `--env` is ignored in the local tracker. Let's make a decision whether to honor it in other trackers.

Some considerations:
* Originally, only the YARN tracker used to honor `--env` argument; other trackers used to ignore it.
* This pull request changes the SSH tracker to recognize `--env`.

It actually may be easier to ignore `--env` in all trackers other than the YARN tracker. This retains the original semantics, and for the local tracker user can always manually set env vars.

Reply:
My proposal: 1) Reserve `--env` for the YARN tracker, 2) introduce a new argument `--env-both` to accommodate the needs of the SSH tracker.